custom/covid/management/commands/update_cases.py

```diff
@@ -13,58 +18,66 @@ class CaseUpdateCommand(BaseCommand):
     """
 
     def __init__(self):
-        self.output_file = None
         self.extra_options = {}
 
-    def case_block(self):
+    def logger_name(self):
+        """
+        Typically __name__
+        """
         raise NotImplementedError()
 
-    def update_cases(self, domain, case_type, user_id):
+    def case_blocks(self, case):
+        """
+        Return a list of CaseBlock updates, or None if no updates are needed.
+        """
         raise NotImplementedError()
 
-    def find_case_ids_by_type(self, domain, case_type):
-        case_ids = CommCareCase.objects.get_case_ids_in_domain(domain, case_type)
-        print(f"Found {len(case_ids)} {case_type} cases in {domain}")
-        return case_ids
-
-    def log_data(self, domain, command, case_type, total_cases, num_updated, errors, loc_id=None):
-        if self.output_file is not None:
-            with open(self.output_file, "a") as output_file:
-                num_case_updated_str = "{} {}: Updated {} out of the {} {} cases".format(domain, command,
-                                                                                         num_updated, total_cases,
-                                                                                         case_type)
-                if loc_id is not None:
-                    num_case_updated_str += f" in this location:{loc_id}"
-                output_file.write(num_case_updated_str + '\n')
-                for error in errors:
-                    output_file.write(domain + ": " + error + '\n')
+    def find_case_ids(self, domain):
+        return CommCareCase.objects.get_case_ids_in_domain(domain, self.case_type)
 
     def add_arguments(self, parser):
         parser.add_argument('domain')
         parser.add_argument('case_type')
         parser.add_argument('--username', type=str, default=None)
         parser.add_argument('--and-linked', action='store_true', default=False)
-        parser.add_argument('--output-file', type=str, default=None)
+        parser.add_argument('--throttle-secs', type=float, default=0)
 
     def handle(self, domain, case_type, **options):
+        # logger.debug will record something to a file but not print it
+        self.logger = logging.getLogger(self.logger_name())
+        if not settings.UNIT_TESTING:
+            self.logger.addHandler(logging.FileHandler(self.logger_name().split(".")[-1] + ".txt"))
+        self.logger.setLevel(logging.DEBUG)
```

Comment:
you could also make `self.logger` a `@cached_property` to help keep `handle` clean